test/unit/job_test.rb

```diff
@@ -354,6 +354,39 @@
     end
   end
 
+  describe 'Handling date_as_argument persistence in Redis' do
+    before do
+      @args = {
+        name: 'Test',
+        cron: '* * * * *',
+        queue: 'default',
+        klass: 'CronTestClass'
+      }
+    end
+
+    it 'stores date_as_argument in Redis when true' do
+      Sidekiq::Cron::Job.create(@args.merge(date_as_argument: true))
+      stored_job = Sidekiq::Cron::Job.find(@args[:name])
+      assert_equal true, stored_job.date_as_argument?
+    end
+
+    it 'does not store date_as_argument in Redis when false' do
```

Comment:
I believe it actually stores the value

```suggestion
    it 'stores date_as_argument in Redis when false' do
```

However, if we want to check the actual value in redis, it's better to make a redis query instead of `Sidekiq::Cron::Job.find`

```ruby
hash_from_redis = Sidekiq.redis do | conn|
  conn.hgetall("cron_job:default:Test")
end

assert_equal "0", hash_from_redis["date_as_argument"]
```

Reply:
@freemanoid @markets 

Is it a good idea to hard-code the Redis key `cron_job:default:Test`? Changes in future updates could alter the key format, causing the test to break. Instead, we could use 
```ruby
stored_job = Sidekiq::Cron::Job.find(@args[:name])
```
to avoid directly referencing the Redis key.

Reply:
My preference is to make tests stupid and not reuse much code in tests.